.travis.yml

```diff
@@ -13,19 +13,25 @@ env:
   - DOCKER=on LOCAL_NAME=docker-4dn-hic_local REMOTE_NAME=duplexa/4dn-hic CWL_NAME_LIST="hi-c-processing-pairs hi-c-processing-bam addfragtopairs pairsam-filter pairsam-markasdup merge-pairs pairsam-merge pairsam-parse-sort extract-mcool-normvector-for-juicebox cool2mcool bwa-mem"
 
 install:
+  - sudo pip install schema_salad
   - sudo pip install git+https://github.com/SooLee/cwltool.git  # cwltool
 script:
   - if [ "$DOCKER" == "on" ]; then
       docker build -t $LOCAL_NAME .;
       source tests/tests.sh;
     fi;
+    if [ "$TRAVIS_PULL_REQUEST" != false ]; then
+       docker tag -t $LOCAL_NAME REMOTE_NAME:$TRAVIS_PULL_REQUEST_BRANCH;
+    else
+       docker tag -t $LOCAL_NAME REMOTE_NAME:$TRAVIS_BRANCH;
+    fi
```

Comment:
re: https://travis-ci.org/4dn-dcic/docker-4dn-hic/builds/361366423 , would try the following to get multi-line execution + debug indiv commands (using `|` in YAML signifies a block of test, apparently):
```yml
  script:
  - |
    if [ "$DOCKER" == "on" ]; then
       docker build -t $LOCAL_NAME .
       source tests/tests.sh
     fi
  - |
    if [ "$TRAVIS_PULL_REQUEST" != false ]; then
       docker tag -t $LOCAL_NAME REMOTE_NAME:$TRAVIS_PULL_REQUEST_BRANCH
    else
       docker tag -t $LOCAL_NAME REMOTE_NAME:$TRAVIS_BRANCH
    fi
  - |
    for CWL_NAME in $CWL_NAME_LIST; do
        source tests/tests_cwl.sh $CWL_NAME
    done
```
or potentially if gets bigger to put into separate ~ `do_docker_build.sh` bash script

Reply:
Thanks for the comments, Alex! I merged them into one block because the docker build and tag commands are required for the cwl test. These are Travis-specific, so I prefer leaving them inside `.travis.yml` But I see your points. Let me simplify it a bit.